test/e2e/samples.go

```diff
@@ -0,0 +1,129 @@
+package e2e
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"testing"
+
+	framework "github.com/operator-framework/operator-sdk/pkg/test"
+	operator "github.com/redhat-developer/build/pkg/apis/build/v1alpha1"
+	"github.com/stretchr/testify/require"
+	v1 "k8s.io/api/core/v1"
+)
+
+// SamplesTesting encapsulate the probes executed against sample files in build-operator.
+type SamplesTesting struct {
+	t   *testing.T           // testing instance
+	ctx *framework.TestCtx   // operator-sdk test context
+	f   *framework.Framework // operator-sdk test framework parameters
+	ns  string               // namespace
+}
+
+// SampleFiles group resources employed during each test interaction
+type SampleFiles struct {
+	ClusterBuildStrategy string // ClusterBuildStrategy file path
+	BuildStrategy        string // BuildStrategy file path
+	Build                string // Build file path
+	BuildRun             string // BuildRun  file path
+}
+
+// amendOutputImageURL amend container image URL based on informed image repository.
+func (s *SamplesTesting) amendOuputImageURL(b *operator.Build, imageRepo string) {
+	if imageRepo == "" {
+		return
+	}
+	imageURL := fmt.Sprintf("%s:%s", imageRepo, b.Name)
+	b.Spec.Output.ImageURL = imageURL
+	s.t.Logf("Amended object: name='%s', image-url='%s'", b.Name, imageURL)
+}
+
+// amendOutputSecretRef amend secret-ref for output image.
+func (s *SamplesTesting) amendOutputSecretRef(b *operator.Build, secretName string) {
+	if secretName == "" {
+		return
+	}
+	b.Spec.Output.SecretRef = &v1.LocalObjectReference{Name: secretName}
+	s.t.Logf("Amended object: name='%s', secret-ref='%s'", b.Name, secretName)
+}
+
+// amendSourceSecretRef patch Build source.SecretRef with secret name.
+func (s *SamplesTesting) amendSourceSecretRef(b *operator.Build, secretRef string) {
```

Comment:
I think `secretRef` should be `secretName`.

Reply:
Okay, sure.